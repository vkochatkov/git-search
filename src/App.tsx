import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { Props } from './interfaces/interface';
import { HomePage } from './pages/HomePage/HomePage';
import { ProfilePage } from './pages/ProfilePage/ProfilePage';

import './index.scss';

export const App: React.FC<Props> = () => {
  return (
    <BrowserRouter>
      <div className="container">
        <Switch>
          <Route exact path={'/'} component={HomePage} />
          <Route path={'/profile/:urlName'} component={ProfilePage} />
        </Switch>
      </div>
    </BrowserRouter>
  );
};

export default App;
