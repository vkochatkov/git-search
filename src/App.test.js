import React from 'react';
import { configure } from 'enzyme';
import { render } from '@testing-library/react';
import { App } from './App';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { BrowserRouter, Link } from 'react-router-dom';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

const setUp = ({ store }) => {
  return render(
    <Provider store={store}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>,
  );
};

describe('App component', () => {
  const initialState = { users: [{ id: 1 }] };
  const mockStore = configureStore();
  let store;

  it('should render App', () => {
    store = mockStore(initialState);
    const wrapper = setUp({ store });
    expect(wrapper).toMatchSnapshot();
  });
});
