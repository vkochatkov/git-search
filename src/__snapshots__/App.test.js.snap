// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="formgroup mt-4"
        >
          <input
            class="form-control"
            id="exampleDataList"
            placeholder="Type to search..."
            type="text"
          />
        </div>
        <div
          class="row pt-4"
        >
          <div
            class="col-sm-4 mb-4"
          >
            <div
              class="card"
            >
              <img
                class="card-img-top"
              />
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                />
                <a
                  class="btn btn-primary"
                  href="/profile/undefined"
                >
                  Открыть
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="formgroup mt-4"
      >
        <input
          class="form-control"
          id="exampleDataList"
          placeholder="Type to search..."
          type="text"
        />
      </div>
      <div
        class="row pt-4"
      >
        <div
          class="col-sm-4 mb-4"
        >
          <div
            class="card"
          >
            <img
              class="card-img-top"
            />
            <div
              class="card-body"
            >
              <h5
                class="card-title"
              />
              <a
                class="btn btn-primary"
                href="/profile/undefined"
              >
                Открыть
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
